src/common/primitive_cache.cpp

```diff
@@ -28,6 +28,7 @@
 #endif
 
 #include <algorithm>
+#include <thread>
```

Comment:
The `thread` header is already included in `primitive_cache_t.hpp`:
https://github.com/oneapi-src/oneDNN/blob/43a7b985c243d7fdaa3f3d5bf9905a37687ea745/src/common/primitive_cache.hpp#L21-L23

We have fixed the GCC 11 related issues you are referring to some time ago, master contains all the fixes.